src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets

```diff
@@ -227,6 +229,19 @@ Copyright (c) .NET Foundation. All rights reserved.
     
   </Target>
 
+  <!-- Throws an error when currently used runtime identifier is not present in project.assets.json. -->
+  <Target Name="CheckRuntimeIdentifier"
+          Condition="'$(DesignTimeBuild)' != 'true' And Exists('$(ProjectAssetsFile)')"
```

Comment:
MSBuild lets you query environment variables as if they were variables in MSBuild. Generally when we add a new warning, we add a flag enviroment variable to disable the warning/error in case it breaks customers unexpectedly. I would add one here such as `DOTNET_CLI_DISABLE_RUNTIME_ASSETS_RESTORE_CHECK`.